parlai/mturk/core/react_server/dev/components/core_components.jsx

```diff
@@ -822,11 +822,140 @@ class TextResponse extends React.Component {
   }
 }
 
+class FormResponse extends React.Component {
+  // Provide a form-like interface to MTurk interface.
+
+  constructor(props) {
+    super(props);
+    // At this point it should be assumed that task_data
+    // has a field "respond_with_form"
+    let responses = []
+    for (let _ of this.props.task_data["respond_with_form"]){
+      responses.push('');
+    }
+    this.state = {'responses': responses, 'sending': false};
+  }
+
+
+  tryMessageSend() {
+    let form_elements = this.props.task_data["respond_with_form"];
+    let response_data = {};
+    let response_text = "";
+    for (let ind in form_elements){
+      let question = form_elements[ind]["question"];
+      let response = this.state.responses[ind];
+      response_data[question] = response;
```

Comment:
Rather than keying on the displayed `question` text, it might make more sense for users to be able to provide their own keys when they specify the question in their config. This would also give you a key that you can use for the form element's key.